### PR TITLE
chore: bump kubernetes versions

### LINKED ISSUE #2558: Kubernetes-based suites are failing
Build URL:  https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/7.17/691/pipeline

Happening when running Helm and k8s-autodiscover suites.

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
This PR aligns the kubernetes version, matching it with current k8s version in Beats, as pointed out by @ChrsMark: https://github.com/elastic/beats/blob/ea207346d651448b8917b0791b2b117b9f9b9212/deploy/kubernetes/Jenkinsfile.yml#L21

Besides that, the PR fixes the path used in the install-test-dependencies script, so that the right path is evaluated. In the past, it used the current dir, but in the cloud instances, it's incorrect.

Another change is in how k8s resources are retrieved: since v1.20, the `selfLink` attribute is not present anymore (see https://github.com/kubernetes/enhancements/issues/1164), so we are updating how the tests are retrieving the selector of a resource: `--selector=app=apm-server`

A side effect we found here is that the `Endpoint` integration has been renamed to `Endpoint and Cloud Security`. @IgorGuz2000 are you aware of that change?

Finally, it bumps the elastic helm charts from 7.11 to 7.17.3 💥  I wonder if this can be automated too (cc/ @jmlrt @v1v)

## Why is it important?
Kubernetes 1.18 has been out of support already.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ] @ChrsMark should we bump kind and Helm versions too? What versions should we use?

## Related issues

- Closes #2558
- Closes #2557

## Follow-ups
We could automate the bump of the k8s version at the moment Beats updates it.

### CODE REVIEW

File: .ci/scripts/install-test-dependencies.sh [hunk 1]
```diff
@@ -11,20 +11,20 @@ set -euxo pipefail
 # Parameters:
 #   - SUITE - that's the name of the test suite to install the dependencies for.
 #
-
+BASE_DIR="${BASE_DIR:-$(pwd)}"
```
Comment (mdelapenya) [inline .ci/scripts/install-test-dependencies.sh:14]: See https://github.com/elastic/e2e-testing/pull/2562/files#r881577813

**Comment by mdelapenya** [comment]
@ChrsMark I see this error only in the scenario for ` kubernetes-autodiscover for filebeat`

Build URL: https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/PR-2562/1/pipeline/720

```
[2022-05-25T13:14:14.964Z] --- Failed steps:
[2022-05-25T13:14:14.964Z] 
[2022-05-25T13:14:14.964Z]   Scenario: Logs collection from a pod with an ephemeral container # features/filebeat.feature:32
[2022-05-25T13:14:14.964Z]     When an ephemeral container is started in "a pod" # features/filebeat.feature:36
[2022-05-25T13:14:14.964Z]       Error: failed to create ephemeral container: exit status 1. Is EphemeralContainers feature flag enabled in the cluster?
```

Error: 
> failed to create ephemeral container: exit status 1. Is EphemeralContainers feature flag enabled in the cluster?

Besides that, the Helm test suite fails for APM Server and for Metricbeat:

**APM Server:**
```
[2022-05-25T13:01:05.667Z]   Scenario: The APM Server chart will create recommended K8S resources # features/apm_server.feature:5
[2022-05-25T13:01:05.667Z]     Then a "Deployment" will manage the pods # features/apm_server.feature:8
[2022-05-25T13:01:05.667Z]       Error: exit status 1
```
> [2022-05-25T12:59:04.688Z] ERRO[2022-05-25T12:59:04Z] Error executing command                       args="[get --raw /scale]" baseDir=. command=kubectl env="map[]" error="exit status 1" stderr="Error from server (NotFound): the server could not find the requested resource\n"

**Metricbeat:**
```
[2022-05-25T13:07:42.833Z]   Scenario: The Metricbeat chart will create recommended K8S resources # features/metricbeat.feature:5
[2022-05-25T13:07:42.833Z]     When the "metricbeat" Elastic's helm chart is installed # features/metricbeat.feature:7
[2022-05-25T13:07:42.833Z]       Error: exit status 1
```

> [2022-05-25T13:05:56.192Z] ERRO[2022-05-25T13:05:55Z] Error executing command                       args="[install metricbeat elastic/metricbeat --version 7.11.2]" baseDir=. command=helm env="map[]" error="exit status 1" stderr="Error: unable to build kubernetes objects from release manifest: [unable to recognize \"\": no matches for kind \"ClusterRole\" in version \"rbac.authorization.k8s.io/v1beta1\", unable to recognize \"\": no matches for kind \"ClusterRoleBinding\" in version \"rbac.authorization.k8s.io/v1beta1\"]\n"

It seems related to the other versions: Helm, kind... what versions do you propose?

**Comment by mdelapenya** [comment]
Bumping Kind version to [0.12.0](https://github.com/kubernetes-sigs/kind/releases/tag/v0.12.0), which is the one compatible with k8s 1.23.4

**Comment by mdelapenya** [comment]
> ```
> [2022-05-25T13:14:14.964Z] --- Failed steps:
> [2022-05-25T13:14:14.964Z] 
> [2022-05-25T13:14:14.964Z]   Scenario: Logs collection from a pod with an ephemeral container # features/filebeat.feature:32
> [2022-05-25T13:14:14.964Z]     When an ephemeral container is started in "a pod" # features/filebeat.feature:36
> [2022-05-25T13:14:14.964Z]       Error: failed to create ephemeral container: exit status 1. Is EphemeralContainers feature flag enabled in the cluster?
> ```

I've checked that `ephemeralContainers` is enabled in the `kind.yml` file: https://github.com/elastic/e2e-testing/blob/main/e2e/_suites/kubernetes-autodiscover/testdata/kind.yml#L4 @ChrsMark any guess?

**Review by kuisathaverat — APPROVED** [review]

**Comment by ChrsMark** [comment]
> > ```
> > [2022-05-25T13:14:14.964Z] --- Failed steps:
> > [2022-05-25T13:14:14.964Z] 
> > [2022-05-25T13:14:14.964Z]   Scenario: Logs collection from a pod with an ephemeral container # features/filebeat.feature:32
> > [2022-05-25T13:14:14.964Z]     When an ephemeral container is started in "a pod" # features/filebeat.feature:36
> > [2022-05-25T13:14:14.964Z]       Error: failed to create ephemeral container: exit status 1. Is EphemeralContainers feature flag enabled in the cluster?
> > ```
> 
> I've checked that `ephemeralContainers` is enabled in the `kind.yml` file: https://github.com/elastic/e2e-testing/blob/main/e2e/_suites/kubernetes-autodiscover/testdata/kind.yml#L4 @ChrsMark any guess?

This `Error: failed to create ephemeral container: exit status 1. Is EphemeralContainers feature flag enabled in the cluster?` seems to be the main issue. Did you try to interact with that kind cluster live and see if you can create ephemeral containers? For some reason even if the feature is enabled it seems that it is not allowed 🤔  no idea why though.

**Comment by ChrsMark** [comment]
Running locally with 

```TAGS="filebeat" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE DEVELOPER_MODE=true ELASTIC_APM_ACTIVE=false PROVIDER=docker make -C e2e/_suites/kubernetes-autodiscover functional-test ```


I get ```ERRO[2022-05-30T15:16:11+01:00] Error executing command                       args="[--kubeconfig /var/folders/vm/_dgw8f996bsfxx5gx0tjjcvm0000gn/T/test-4160273331/kubeconfig --namespace test-013a14fc-ab01-4fc5-b339-df80d55c59f1 replace --raw /api/v1/namespaces/test-013a14fc-ab01-4fc5-b339-df80d55c59f1/pods/a-pod/ephemeralcontainers -f -]" baseDir=. command=kubectl env="map[]" error="exit status 1" stderr="Error from server (BadRequest): EphemeralContainers in version \"v1\" cannot be handled as a Pod: no kind \"EphemeralContainers\" is registered for version \"v1\" in scheme \"k8s.io/kubernetes/pkg/api/legacyscheme/scheme.go:30\"\n"```.


So it seems that the definition of the ephemeral container at https://github.com/elastic/e2e-testing/blob/fb6385e1732e3f1399da00d40e0504c4bfe74159/e2e/_suites/kubernetes-autodiscover/autodiscover_test.go#L256-L276 is not valid anymore. 

Indeed it seems that things have changed in Kubernetes side:
 https://v1-21.docs.kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/
vs
https://v1-23.docs.kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/ (similar at v1.22)

We need to find a way to define/add ephemeral containers for this new version of k8s. I'm on it.

**Comment by mdelapenya** [comment]
@ChrsMark feel free to push to this branch, so that it's possible to use this PR to fix the build

Thanks!

**Comment by ChrsMark** [comment]
> @ChrsMark feel free to push to this branch, so that it's possible to use this PR to fix the build
> 
> Thanks!

@mdelapenya I have a fix that worked on my end. Can you cherry pick this from https://github.com/elastic/e2e-testing/pull/2577? I'm not sure if I can directly push to your branch.

**Comment by mdelapenya** [comment]
@ChrsMark I think it fixed the k8s-autodiscover tests! Great!

What I see is that the helm chart scenario for apm-server is failing with:

```
[2022-05-30T16:48:22.916Z] ERRO[2022-05-30T16:48:18Z] Error executing command                       args="[get --raw /scale]" baseDir=. command=kubectl env="map[]" error="exit status 1" stderr="Error from server (NotFound): the server could not find the requested resource\n"
[2022-05-30T16:48:22.916Z] TRAC[2022-05-30T16:48:19Z] After Helm scenario: The APM Server chart will create recommended K8S resources 
[2022-05-30T16:49:01.677Z]     Then a "Deployment" will manage the pods                                             # helm_charts_test.go:224 -> *HelmChartTestSuite
[2022-05-30T16:49:01.677Z]     exit status 1
```

**Comment by jmlrt** [comment]
> Finally, it bumps the elastic helm charts from 7.11 to 7.17.3 💥 I wonder if this can be automated too (cc/ @jmlrt @v1v)

What kind of automation do you have in mind? Automating version bump in the e2e-testing codebase should be easy, however, If you want it to be triggered automatically when a new Helm release is done, I'm not sure what's the best way? Maybe it's possible to use a GitHub webhook event when a new tag or release is created in the helm-charts repo.

File: e2e/_suites/fleet/features/backend_processes.feature [hunk 2]
```diff
@@ -71,5 +71,5 @@ Scenario Outline: Un-enrolling Elastic Agent stops Elastic Endpoint
 @deploy-endpoint-then-remove-it-from-policy
 Scenario Outline: Removing Endpoint from Agent policy stops the connected Endpoint
   Given an "Endpoint" is successfully deployed with an Agent using "tar" installer
-  When the "Endpoint Security" integration is "removed" in the policy
+  When the "Endpoint and Cloud Security" integration is "removed" in the policy
```
Comment (mdelapenya) [inline e2e/_suites/fleet/features/backend_processes.feature:74]: I wonder how this has not been found before 👀

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**